src/main/java/io/vertx/core/http/impl/pool/Pool.java

```diff
@@ -346,6 +350,7 @@ private synchronized void setConcurrency(Holder holder, long concurrency) {
       throw new IllegalArgumentException("Cannot set a negative concurrency value");
     }
     if (holder.removed) {
+      assert false : "Cannot recycle removed holder";
```

Comment:
Do we want to keep this assertion? We don't use them afaik

Reply:
it is actually used in tests (enabled when running tests) which is very useful

Reply:
I mean that it can catch more bugs

Reply:
(and it did :-) )